pennylane/tape/qscript.py

```diff
@@ -552,33 +551,7 @@ def trainable_params(self, param_indices):
 
         self._trainable_params = sorted(set(param_indices))
 
-    def get_operation(self, idx, return_op_index=False):
```

Comment:
Can we instead just toggle `return_op_index=True` as the default and remove the warning?

If we are removing the `return_op_index=False` keyword argument, we will also need to make PR's to lightning, lightning-gpu, and lightning-kokkos, as I just modified them to request `return_op_index=True` so that it eliminated the warning. If we remove the keyword argument, we will need to go back through and change all usages of them.

Reply:
That's not what was the plan and in general it is not very good for the signature to depend on the arguments. I have already opened for ligntning, I will do the same for gpu and kokkos.